src/transaction/log_manager.c

```diff
@@ -3735,9 +3735,11 @@ log_append_assigned_mvccid (THREAD_ENTRY * thread_p, MVCCID mvccid)
 }
 
 void
-log_append_schema_modification_lock (THREAD_ENTRY * thread_p, const OID * classoid)
+log_append_locked_object (THREAD_ENTRY * thread_p, const OID * classoid, const OID * oid, const LOCK lock)
 {
   assert (!OID_ISNULL (classoid));
+  assert (!OID_IS_ROOTOID (classoid) || lock == SCH_M_LOCK);	// if classoid is rootoid, lock must be SCH_M_LOCK
+  assert (!OID_EQ (classoid, oid_Serial_class_oid) || lock == X_LOCK);	// if classoid is serial class oid, lock must be X_LOCK
```

Comment:
Are these 2 asserts equivalent to the conditions used to call this function?
```
		  const bool is_lock_for_class = OID_IS_ROOTOID (class_oid) && lock == SCH_M_LOCK;
		  const bool is_lock_for_serial = OID_EQ (class_oid, oid_Serial_class_oid) && lock == X_LOCK;
		  if (is_lock_for_class || is_lock_for_serial)
```

As they are written, each one can be valid in the negative form - when the class oid is neither root class nor serial.
As the function is called, at least one of them has to be valid in the affirmative form.

Reply:
Do you mean that additional assertion is required ? 
```
assert (is_lock_for_class || is_lock_for_serial); 
```

Reply:
Yes, something analogue to that.